pkg/hubble/exporter/config.go

```diff
@@ -0,0 +1,237 @@
+			continue
+		}
+		name := c.FlowLogs[i].Name
+		if name == "" {
+			errs = errors.Join(errs, fmt.Errorf("name is required"))
+		} else {
+			if _, ok := flowlogNames[name]; ok {
+				errs = errors.Join(errs, fmt.Errorf("duplicated flowlog name %s", name))
+			}
+			flowlogNames[name] = struct{}{}
+		}
+
+		filePath := c.FlowLogs[i].FilePath
+		if filePath == "" {
+			errs = errors.Join(errs, fmt.Errorf("filePath is required"))
+		} else {
+			if _, ok := flowlogPaths[filePath]; ok {
+				errs = errors.Join(errs, fmt.Errorf("duplicated flowlog path %s", filePath))
+			}
+			flowlogPaths[filePath] = struct{}{}
+		}
+	}
+
+	return errs
+}
+
+var _ ExporterConfig = (*FlowLogConfig)(nil)
+
+// FlowLogConfig represents configuration of single dynamic exporter.
+type FlowLogConfig struct {
+	Name           string      `json:"name,omitempty" yaml:"name,omitempty"`
+	FilePath       string      `json:"filePath,omitempty" yaml:"filePath,omitempty"`
+	FieldMask      FieldMask   `json:"fieldMask,omitempty" yaml:"fieldMask,omitempty"`
+	IncludeFilters FlowFilters `json:"includeFilters,omitempty" yaml:"includeFilters,omitempty"`
+	ExcludeFilters FlowFilters `json:"excludeFilters,omitempty" yaml:"excludeFilters,omitempty"`
+	FileMaxSizeMB  int         `json:"fileMaxSizeMb,omitempty" yaml:"fileMaxSizeMb,omitempty"`
+	FileMaxBackups int         `json:"fileMaxBackups,omitempty" yaml:"fileMaxBackups,omitempty"`
+	FileCompress   bool        `json:"fileCompress,omitempty" yaml:"fileCompress,omitempty"`
+	End            *time.Time  `json:"end,omitempty" yaml:"end,omitempty"`
+}
+
+// Equal implements ExporterConfig.
+func (f *FlowLogConfig) Equal(other any) bool {
+	if other, ok := other.(*FlowLogConfig); ok {
+		return f.equals(other)
+	}
+	return false
+}
+
+// Equal implements ExporterConfig.
+func (f *FlowLogConfig) IsActive() bool {
+	return f.End == nil || f.End.After(time.Now())
+}
+
+func (f *FlowLogConfig) equals(other *FlowLogConfig) bool {
+	if f == nil && other == nil {
+		return true
+	}
+	if f == nil || other == nil {
+		return false
+	}
+
+	if f.FilePath != other.FilePath {
+		return false
+	}
+
+	if !f.FieldMask.equals(other.FieldMask) {
+		return false
+	}
+
+	if !f.IncludeFilters.equals(other.IncludeFilters) {
+		return false
+	}
+
+	if !f.ExcludeFilters.equals(other.ExcludeFilters) {
+		return false
+	}
+
+	if f.FileMaxSizeMB != other.FileMaxSizeMB {
+		return false
+	}
+
+	if f.FileMaxBackups != other.FileMaxBackups {
+		return false
+	}
+
+	if f.FileCompress != other.FileCompress {
+		return false
+	}
+
+	if f.End == nil && other.End != nil ||
+		f.End != nil && other.End == nil ||
+		f.End != nil && other.End != nil && !f.End.Equal(*other.End) {
+		return false
+	}
+
+	return true
+}
+
+// FlowFilters is a slice of filters to include or exclude flows.
+type FlowFilters []*flowpb.FlowFilter
+
+func (f FlowFilters) equals(other FlowFilters) bool {
+	aFiltersSet, bFiltersSet := make(map[string]bool), make(map[string]bool)
```

Comment:
nit: we know the needed map sizes
```suggestion
	aFiltersSet := make(map[string]bool, len(f))
	bFiltersSet := make(map[string]bool, len(other))